src/libs/SubscriptionUtils.ts

```diff
@@ -30,25 +120,220 @@ Onyx.connect({
     waitForCollectionCallback: true,
 });
 
-let ownerBillingGraceEndPeriod: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END,
-    callback: (value) => (ownerBillingGraceEndPeriod = value),
-});
-
-let amountOwed: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_AMOUNT_OWNED,
-    callback: (value) => (amountOwed = value),
-});
-
 let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     callback: (value) => (allPolicies = value),
     waitForCollectionCallback: true,
 });
 
+/**
+ * @returns The date when the grace period ends.
+ */
+function getOverdueGracePeriodDate(): OnyxEntry<number> {
+    return ownerBillingGraceEndPeriod;
+}
+
+/**
+ * @returns Whether the workspace owner has an overdue grace period.
+ */
+function hasOverdueGracePeriod(): boolean {
+    return !!ownerBillingGraceEndPeriod ?? false;
+}
+
+/**
+ * @returns Whether the workspace owner's grace period is overdue.
+ */
+function hasGracePeriodOverdue(): boolean {
+    return !!ownerBillingGraceEndPeriod && Date.now() < new Date(ownerBillingGraceEndPeriod).getTime();
+}
+
+/**
+ * @returns The amount owed by the workspace owner.
+ */
+function getAmountOwed(): number {
+    return amountOwed ?? 0;
+}
+
+function hasAmountOwed(): boolean {
+    return !!amountOwed;
+}
+
+/**
+ * @returns Whether there is a card authentication error.
+ */
+function hasCardAuthenticatedError() {
+    return stripeCustomerId?.status === 'authentication_required' && amountOwed === 0;
+}
+
+/**
+ * @returns Whether there is a billing dispute pending.
+ */
+function hasBillingDisputePending() {
+    return !!billingDisputePending ?? false;
+}
+
+/**
+ * @returns Whether there is a card expired error.
+ */
+function hasCardExpiredError() {
+    return billingStatus?.declineReason === 'expired_card' && amountOwed !== 0;
+}
+
+/**
+ * @returns Whether there is an insufficient funds error.
+ */
+function hasInsufficientFundsError() {
+    return billingStatus?.declineReason === 'insufficient_funds' && amountOwed !== 0;
+}
+
+/**
+ * @returns The card to be used for subscription billing.
+ */
+function getCardForSubscriptionBilling(): Fund | undefined {
+    return Object.values(fundList ?? {}).find((card) => card?.isDefault);
+}
+
+/**
+ * @returns Whether the card is due to expire soon.
+ */
+function hasCardExpiringSoon(): boolean {
+    if (!isEmptyObject(billingStatus ?? {})) {
+        return false;
+    }
+
+    const card = getCardForSubscriptionBilling();
+
+    if (!card) {
+        return false;
+    }
+
+    return card?.accountData?.cardYear === new Date().getFullYear() && card?.accountData?.cardMonth === new Date().getMonth() + 1;
```

Comment:
Wouldn't this break when we are in December and the card is expiring January next year?

Reply:
Also what happens when the card is expiring in the current month?

Reply:
```
 const isExpiringThisMonth = cardYear === currentYear && cardMonth === currentMonth;
            const isExpiringNextMonth = cardYear === (currentMonth === 12 ? currentYear + 1 : currentYear) 
                                         && cardMonth === (currentMonth === 12 ? 1 : currentMonth + 1);

            return isExpiringThisMonth || isExpiringNextMonth;
```